src/main/java/eu/europeana/api/recommend/service/RecommendService.java

```diff
@@ -75,11 +84,151 @@ public Mono getRecommendationsForRecord(String recordId, int pageSize, String to
+     * 4) if there are no results from any of the above api : empty request will be formed and sent to engine
+     * ex: {"labels":[{"title":"","descriptions":null}],"items":[""]}
+     * This is to avoid Internal Server errors from engine due to null postBody
+     *
+     * @param entityId
+     * @param apikey
+     * @return
+     */
+    public String getEntityRecommendationRequest(String entityId, String apikey) {
+        EntityRecommendRequest request = new EntityRecommendRequest();
+
+        // fetch the skos:prefLabels and skos:altlabels from Entity API
+        StringBuilder entityApiUrl = new StringBuilder(config.getEntityApiEndpoint())
+                .append(getRecommendServiceUtils().entityApiSearchQuery(entityId, apikey));
+        getLabels(request, entityApiUrl.toString());
+
+        // fetch Entity Set items from set api
+        StringBuilder setApiUrl = new StringBuilder(config.getSetApiEndpoint()).
+                append(getRecommendServiceUtils().setApiSearchQuery(entityId, config.getSetApiPageSize(), apikey));
+        getItems(request, setApiUrl.toString());
+
+        // serialise the request
+        return serialiseEntityRequest(request);
+    }
+
+    /**
+     * Sets the labels in the EntityRecommendRequest
+     * extracts the prefLabel and altLabel from entity api json response
+     *
+     * @param request
+     * @param entityUrl
+     */
+    private void getLabels(EntityRecommendRequest request, String entityUrl) {
+        try {
+            JSONObject jsonObject = new JSONObject(getEntityApiSearchResponse(entityUrl));
+            List<String> extractedLabels = getRecommendServiceUtils().extractLabels(jsonObject);
+            List<Labels> labels = new ArrayList<>();
+            if (!extractedLabels.isEmpty()) {
+                for (String label : extractedLabels) {
+                    labels.add(new Labels(label));
+                }
+                request.setLabels(labels);
+            } else {
+                labels.add(new Labels(""));
+                request.setLabels(labels);
+            }
+        } catch (JSONException e) {
+            LOG.error("Error parsing entity api response for url {}", entityUrl);
+        }
+    }
+
+    /**
+     * Sets the items in the EntityRecommendRequest.
+     * Extracts the items from the set api json response
+     *
+     * @param request
+     * @param setApiUrl
+     */
+    private void getItems(EntityRecommendRequest request, String setApiUrl) {
+        try {
+            JSONObject jsonObject = new JSONObject(getSetApiSearchResponse(setApiUrl));
+            List<String> items=getRecommendServiceUtils().extractItems(jsonObject);
+            if (!items.isEmpty()) {
+                request.setItems(items.toArray(new String[0]));
+            } else {
+                request.setItems(new String[]{""});
+            }
+        } catch (JSONException e) {
+            LOG.error("Error parsing set api response for url {}", setApiUrl);
```

Comment:
Can we also log the JSONException as well (and fix the SQ warning that we're ignoring it)?

Reply:
logged exception as well